x-pack/plugins/observability_onboarding/public/components/app/custom_logs/wizard/install_elastic_agent.tsx

```diff
@@ -57,13 +56,20 @@ export function InstallElasticAgent() {
   }
 
   function onAutoDownloadConfig() {
-    const { autoDownloadConfig, ...state } = getState();
-    setState({ ...state, autoDownloadConfig: !autoDownloadConfig });
+    setState((state) => ({
+      ...state,
+      autoDownloadConfig: !state.autoDownloadConfig,
+    }));
   }
 
   const { data: monitoringRole, status: monitoringRoleStatus } = useFetcher(
     (callApi) => {
-      if (CurrentStep === InstallElasticAgent) {
+      const { apiKeyEncoded, onboardingId } = getState();
+      if (
+        CurrentStep === InstallElasticAgent &&
+        !apiKeyEncoded &&
```

Comment:
shouldn't this be `!(apiKeyEncodded && !onboardingId)`?
Maybe i'm not seeing something but if any of those if not present we should call `POST /internal/observability_onboarding/custom_logs/save` which will need the user to have the right privileges, right?

Reply:
Should we then sync [this](https://github.com/elastic/kibana/blob/246fdf3d33777cdd339e553d6860ab70116e8b33/x-pack/plugins/observability_onboarding/public/components/app/custom_logs/wizard/install_elastic_agent.tsx#L107) then? It's a bit confusing